File: ios/event/LocationEvent.m
//
//  LocationEvent.m
//  RNBackgroundLocation
//
//  Location Event
//  Android LocationEvent.java benzeri
//

#import "LocationEvent.h"
#import "LocationModel.h"

@implementation LocationEvent

- (instancetype)initWithLocation:(LocationModel *)location {
    self = [super init];
    if (self) {
        _location = location;
    }
    return self;
}

- (NSDictionary *)toDictionary {
    return [self.location toDictionary];
}

- (NSString *)getEventName {
    return @"location";
}

@end

File: android/src/main/java/com/backgroundlocation/event/LocationEvent.java
package com.backgroundlocation.event;

import org.json.JSONObject;

/**
 * Location Event
 * Type-safe event class
 */
public class LocationEvent {
    private final JSONObject locationJson;
    
    public LocationEvent(JSONObject locationJson) {
        this.locationJson = locationJson;
    }
    
    public JSONObject toJson() {
        return locationJson;
    }
    
    public String getEventName() {
        return "location";
    }
}



